### PR TITLE
preserve json parsing error

### PR DESCRIPTION
**Pull request by bcoca**
* preserve json parsing error
* added test

(cherry picked from commit bbdf77a59f50306b34161efc148b572ab257a537)

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

dataloader

### CODE REVIEW

**Review by Andersson007 — COMMENTED** [review]
There are unit test for every file. Would be better to add checks according to the changes.

File: lib/ansible/parsing/utils/yaml.py [hunk 1]
```diff
@@ -36,9 +36,10 @@ def _handle_error(yaml_exc, file_name, show_content):
         err_obj = AnsibleBaseYAMLObject()
         err_obj.ansible_pos = (file_name, yaml_exc.problem_mark.line + 1, yaml_exc.problem_mark.column + 1)
 
-    err_msg = getattr(yaml_exc, 'problem', '')
+    err_msg = 'We were unable to read either as JSON nor YAML, these are the errors we got from each:\n' \
```
Comment (samdoran) [inline lib/ansible/parsing/utils/yaml.py:39]:
```suggestion
    err_msg = 'Unable to read as JSON or YAML.\n\n' \
```

File: lib/ansible/parsing/utils/yaml.py [hunk 2]
```diff
@@ -36,9 +36,10 @@ def _handle_error(yaml_exc, file_name, show_content):
...
+              'JSON: %s\n\n' % to_text(json_exc) + YAML_SYNTAX_ERROR % getattr(yaml_exc, 'problem', '')
```
Comment (samdoran) [inline lib/ansible/parsing/utils/yaml.py:40]:
```suggestion
              'Syntax Error while loading JSON.\n  %s\n\n' % to_text(json_exc) + YAML_SYNTAX_ERROR % getattr(yaml_exc, 'problem', '')
```

Comment (samdoran) [reply lib/ansible/parsing/utils/yaml.py:40]: Maybe make the JSON error message match `YAML_SYNTAX_ERROR`. That way the output is more uniform:
```
ERROR! Unable to load as JSON or YAML.

Syntax Error while loading JSON:
  Expecting value: line 1 column 1 (char 0)

Syntax Error while loading YAML.
  mapping values are not allowed in this context
...
```

**Review by samdoran — COMMENTED** [review]
The overall change looks good. We need a changelog and unit tests.

I made a few suggestions on the message and formatting, but those aren't blockers.

**Comment by AlanCoding** [comment]
> (cherry picked from commit bbdf77a)

What do you mean by this? _this_ PR targeted and merged into `devel`. The backport PR was 69282. Where was this cherry-picking from?

**Pull request merged**